File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/api/IssuanceApi.kt
package my.ssdid.sdk.api

import my.ssdid.sdk.domain.oid4vci.CredentialOffer
import my.ssdid.sdk.domain.oid4vci.CredentialOfferReview
import my.ssdid.sdk.domain.oid4vci.IssuanceResult
import my.ssdid.sdk.domain.oid4vci.IssuerMetadata
import my.ssdid.sdk.domain.oid4vci.OpenId4VciHandler

class IssuanceApi internal constructor(private val handler: OpenId4VciHandler) {
    fun processOffer(uri: String): Result<CredentialOfferReview> = handler.processOffer(uri)

    suspend fun acceptOffer(
        offer: CredentialOffer,
        metadata: IssuerMetadata,
        selectedConfigId: String,
        txCode: String? = null,
        walletDid: String,
        keyId: String,
        algorithm: String,
        signer: CredentialSigner
    ): Result<IssuanceResult> = handler.acceptOffer(
        offer, metadata, selectedConfigId, txCode, walletDid, keyId, algorithm
    ) { data -> signer.sign(data) }
}

File: sdk/ios/SsdidCore/Sources/SsdidCore/Api/IssuanceApi.swift
import Foundation

/// Facade for OpenID4VCI credential issuance operations.
public struct IssuanceApi {
    let handler: OpenId4VciHandler

    public func processOffer(uri: String) async throws -> CredentialOfferReview {
        try await handler.processOffer(uri: uri)
    }

    public func acceptOffer(
        offer: CredentialOffer,
        metadata: IssuerMetadata,
        selectedConfigId: String,
        txCode: String? = nil,
        walletDid: String,
        keyId: String,
        algorithm: String,
        signer: @escaping @Sendable (Data) -> Data
    ) async throws -> IssuanceResult {
        try await handler.acceptOffer(
            offer: offer,
            metadata: metadata,
            selectedConfigId: selectedConfigId,
            txCode: txCode,
            walletDid: walletDid,
            keyId: keyId,
            algorithm: algorithm,
            signer: signer
        )
    }
}
